Fix duplicate connection/logs

jyameo opened the pull request:
Fix duplicate app logs on page refresh
Fixes: https://github.com/dart-lang/webdev/issues/2604

Problem
Each page refresh created duplicate stdout listeners for the same app, causing exponential log duplication (1 log → 2 logs → 3 logs, etc.).

Root Cause
app_domain.dart created new VM service stdout listeners on every dwds.connectedApps event without checking if listeners already existed for the same appId.

Solution
Added simple tracking to prevent duplicate listeners:

Track active appIds in a Set<String> _activeListeners
Only create stdout listeners if not already listening for that appId
Clean up tracking when connections end
Changes
app_domain.dart: Added listener deduplication with minimal code changes

Result
 - Single log event per refresh (no more duplicates) 
 - All debugging functionality preserved 
 - No breaking changes

bkonyi commented on webdev/lib/src/daemon/app_domain.dart:
```diff
@@ -121,8 +111,7 @@ class AppDomain extends Domain {
       unawaited(debugConnection.onDone.whenComplete(() {
-        appState.dispose();
-        _appStates.remove(appId);
+        _cleanupAppConnection(appId, appState);
```
Won't this result in the streams being cancelled once the first `DebugConnection` closes, even if a more recent connection to the same `appId` has been established?

bkonyi replied:
Disregard, the `DebugConnection` is shared across connections to the same `appId`.

bkonyi commented on webdev/lib/src/daemon/app_domain.dart:
```diff
@@ -223,6 +220,35 @@ class AppDomain extends Domain {
+      String appId, VmService vmService) async {
+    try {
+      vmService.onServiceEvent.listen(_onServiceEvent);
+      await vmService.streamListen('Service');
```
Nit: replace `'Service'` with `EventStreams.kService`

jyameo commented:
@bkonyi I made the following changes: 
- use `_appStates.containsKey(appId)` to check for existing connections. 
- I removed the _activeListeners set entirely since the check is happening at the AppStates level. 
- Also removed the unnecessary `vmService.streamCancel('Stdout');`
- I wasn't sure about `server.buildResults` so I left it as is for now. 

Let me know what you think and If I missed something here. Thanks :)
